# Gambler's fallacy and probability of independent events

In [1]:
import numpy as np

The Gambler's Fallacy: A false belief that if something happens very frequently in a given period of time, it will happen less frequently in the future. 
Independent events don't work like this. Tossing a coin 100 times won't affect the 101th toss. This notebook is a simple showcase of gambler's policy and probability of independent events.


If events A and B are independent then, 


P(AB) = P(A or B) = P(A) * P(B)


*(P(X) means probability of the event X)*

Let's verify this by rolling a die:

In [2]:
class Dice():
    def __init__(self):
        self.outcome = None
    
    def roll(self, n=1):
        self.outcome = np.random.randint(1, 7, n)
        return self.outcome
    
    def reset(self):
        self.outcome= None

What is the probability of the event A = atleast one *six* in 10 dice rolls?


Mathematically, 
```
P(A) = 1 - P(A')                 (A' is the probability of not getting any six in 10 rolls)
     = 1 - (A1 A2 A3 ..... A10)  (Ai is the probability of not getting six in ith roll)
     = 1 - (5/6)^10
     = 0.83849 or 83.849% 
```
Empirically:

In [3]:
# Let's roll the dice 10 times for a million times
dice = Dice()
sixes = 0
N = int(1e6)
for i in range(N):
    rolls = dice.roll(10)
    if 6 in rolls:
        sixes += 1
P = sixes/1e6
print(f"Probability of atleast 1 six in 10 rolls is: {P}")

Probability of atleast 1 six in 10 rolls is: 0.837983


# Conclusion
The mathematically calculated probability was 0.83849 and empirically it turns out to be 0.837983 which is pretty close
Hence, not getting a six for 9 consecutive turns does not mean a higher chance of getting a six in 10th turn. 

**Moral: Don't fall prey to the gambler's fallacy.**